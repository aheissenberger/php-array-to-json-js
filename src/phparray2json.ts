import spawn from './spawn-promise'

export default async function phparray2json(phparray: string, phpExecPath?: string) {
    const phpCode = '<?php echo json_encode(' + phparray + ');'
    const result: any = await spawn(phpExecPath ? phpExecPath : 'php', [], {}, phpCode);
    return result?.stdout?.toString() ?? ''
}