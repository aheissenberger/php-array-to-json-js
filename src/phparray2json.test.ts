import { describe, test, expect } from 'vitest';
import which from 'which';

import phparray2json from './phparray2json';

const phpExecNotExists = (which.sync('php', { nothrow: true }) === null)

describe.skipIf(phpExecNotExists)('check php result string syntax', () => {
    const phpdata = [
        [`1`, `1`],
        [`"string"`, `"string"`],
        [`true`, `true`],
        [`[]`, `[]`],
        [`["key"=>1]`, `{"key":1}`],
        [`[1,2,3]`, `[1,2,3]`],
        [`["key1"=>1,"key2"=>[1,"string",true]]`, `{"key1":1,"key2":[1,"string",true]}`],
    ];
    test.each(phpdata)('php array: %s => %s', async (input, expected) => {
        const json = await phparray2json(input)
        expect(json).toBe(expected)
    })

    test('broken syntax throws Error', async () => {
        await expect(() => phparray2json(`[`)).rejects.toThrowErrorMatchingSnapshot()
    })

    test('php not found', async () => {
        await expect(() => phparray2json(`1`,'phpnotfound')).rejects.toThrowErrorMatchingSnapshot()
    })
})