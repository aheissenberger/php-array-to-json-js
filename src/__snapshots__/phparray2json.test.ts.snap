// Vitest Snapshot v1

exports[`check php result string syntax > broken syntax throws Error 1`] = `"command exited with code: 255"`;

exports[`check php result string syntax > php not found 1`] = `"Error: spawn phpnotfound ENOENT"`;
